pkg/trace/api/api.go

```diff
@@ -188,29 +205,40 @@ func (r *HTTPReceiver) attachDebugHandlers(mux *http.ServeMux) {
 	mux.Handle("/debug/vars", expvar.Handler())
 }
 
-// Listen creates a new HTTP server listening on the provided address.
-func (r *HTTPReceiver) Listen(addr, logExtra string) error {
-	listener, err := net.Listen("tcp", addr)
-	if err != nil {
-		return fmt.Errorf("cannot listen on %s: %v", addr, err)
+// listenUnix returns a *net.Listener listening on the given "unix" socket path.
+func (r *HTTPReceiver) listenUnix(path string) (net.Listener, error) {
+	fi, err := os.Stat(path)
+	if err == nil {
+		// already exists
+		if fi.Mode()&os.ModeSocket == 0 {
+			return nil, fmt.Errorf("cannot reuse %q; not a unix socket", path)
+		}
+		if err := os.Remove(path); err != nil {
+			return nil, fmt.Errorf("unable to remove stale socket: %v", err)
+		}
 	}
-	ln, err := newRateLimitedListener(listener, r.conf.ConnectionLimit)
```

Comment:
Why use the rate limited listener for Unix but not for TCP?

Reply:
Good question. I actually had to go ahead and try it again only to remember that I did before...

The reason is that the current rate limited listener is bound to [a `*net.TCPListener`](https://github.com/DataDog/datadog-agent/blob/93f76eed5d5c01c5b437a901a1f75efc3deab624/pkg/trace/api/listener.go#L15) which I've tried to change to a regular `net.Listener` only to find out that it won't work because it's using the [`SetDeadline` call](https://github.com/DataDog/datadog-agent/blob/93f76eed5d5c01c5b437a901a1f75efc3deab624/pkg/trace/api/listener.go#L76) from the `*net.TCPListener`.

I guess we could work with making it more generic, but honestly I think that it's a bit risky, maybe not worth it, and possibly for the scope of another PR, given the limited use of UDS. 

The system already has protection against maximum open files which can be set by the user (`uname -a`). Personally I doubt this connection limit adds any value anyway. What do you think?

Reply:
I don't know enough about UDS to judge if it needs rate limiting to be safe, but I think the key question would be can they easily send us too much data too quickly via UDS and if so what are the consequences?  If it's not likely that they'll be able to overload us over UDS then I think started simple makes sense.

There's also the question of what happens when you have both UDS and TCP running: I think you'd have to factor in how much data is coming in through UDS and TCP together to properly determine the rate limit for TCP?

Reply:
From an agent perspective it's fine. We have other, better, rate limiting mechanisms (like the watchdog, based on resource usage). The worst thing that can happen is that they reach their maximum number of open files limit, which is self imposed by the host.

I would consider the current rate limiter to apply only to TCP and would treat UDS as a separate matter.

Reply:
That makes sense to me.